### ISSUE TITLE
is that possible to set set Signature Default from add-in

### ISSUE THREAD

**Issue by junaid042** [OfficeDev/office-js-docs-reference#1930]
is there any way to set signature as default from add-in using office js like every time when I open new mail this mail already have signature no need to open add-in again

**Comment by Rick-Kirkham** [comment]
Yes. See this sample: [Outlook set signature](https://github.com/OfficeDev/Office-Add-in-samples/tree/main/Samples/outlook-set-signature).

**Comment by junaid042** [comment]
I tried this sample yesterday but not working for my in my React/TypeScript Add-in. @Rick-Kirkham any sample in React?

**Comment by junaid042** [comment]
@Rick-Kirkham  simply I want OnMessageCompose get my signature html from localStorage and set is as signature.

**Comment by Rick-Kirkham** [comment]
@junaid042 The Office.js code is the same regardless of whether you are using React for your UI. 

You use the `Office.context.roamingSettings.set()` method to set a signature as show in the sample file [taskpane_main.js](https://github.com/OfficeDev/Office-Add-in-samples/blob/main/Samples/outlook-set-signature/src/taskpane/Js/taskpane_main.js).

The file, [taskpane_render.js](https://github.com/OfficeDev/Office-Add-in-samples/blob/main/Samples/outlook-set-signature/src/taskpane/Js/taskpane_render.js), shows how to read and write from localStorage.

The file, [autorunshared.js](https://github.com/OfficeDev/Office-Add-in-samples/blob/main/Samples/outlook-set-signature/src/runtime/Js/autorunshared.js), shows how to create handlers for events. See the `checkSignature` method.

The 2 manifests files in the sample, manifest.xml and manifest.json, show how to connect the handler to the new message composed event. 

I also recommend that you first read our documentation for event-based add-ins beginning here: [Configure your Outlook add-in for event-based activation](https://learn.microsoft.com/en-us/office/dev/add-ins/outlook/autolaunch).

We use the issues in this repo to track problems with the documentation. Programming questions like this one should be raised on Stack Overflow. I'm going to close this now. If you have further questions, please raise them on Stack Overflow.

**Comment by junaid042** [comment]
@Rick-Kirkham thanks you very much for your help! on stackoverflow no one giving me answer that's why I asked here.
I am trying to finish this and my event is working correctly on web but on Windows desktop when I open new mail this is gives me this error "we can't start this add-in because this is not set up properly" like in this screenshot same and the same thing is working on web outlook any idea? and I have tried to restart my outlook re-installed manifest but it doesn't work.

![image](https://github.com/OfficeDev/office-js-docs-reference/assets/134502135/2e821956-ec24-48df-915e-97b5b8029c46)

**Comment by Rick-Kirkham** [comment]
@junaid042 Create a [Bug Report on the office-js repo](https://github.com/OfficeDev/office-js/issues/new?assignees=&labels=&projects=&template=bug_report.md&title=).

**Comment by junaid042** [comment]
@Rick-Kirkham I created please check it out [issue link](https://github.com/OfficeDev/office-js/issues/4496)
